src/Url.php

```diff
@@ -1066,4 +1066,36 @@ public static function peelUrl($haystack, $prelude_string, $wrapper_start, $wrap
     }
     return FALSE;
   }
+
+  /**
+   * Checks if given URL matches a list of candidates for a default document.
+   *
+   * @param string $url
+   *   The URL to be tested
+   * @param array $candidates
+   *   A list of potential document names that could be indexes.
+   *   Defaults to "default" and "index".
+   *
+   * @return mixed
+   *   string - The base path if a matching document is found.
+   *   bool - FALSE if no matching document is found.
+   */
+  public static function getRedirectIfIndex($url, $candidates = array("default", "index")) {
+    // Assume no match.
+    $root_path = FALSE;
+
+    // Filter through parse_url to separate out querystrings and etc.
+    $path = parse_url($url, PHP_URL_PATH);
+    $filename = strtolower(pathinfo($path, PATHINFO_FILENAME));
```

Comment:
Unfortunately it looks like I led you astray.   On testing this,
$filename = strtolower(pathinfo($path, PATHINFO_FILENAME));   
actually returns a path element if there is no file.
so if $path was path1/default/   where default is the name of the directory, we would get a false positive.
You actually have to look for extension and if you have extension, then $filename is actually a filename.
$extension = pathinfo($path, PATHINFO_EXTENSION);

Then the if would actually look like 
 if (!empty($extension) && !empty($filename) && in_array($filename, $candidates))